librz/arch/isa/msp430/msp430_il.c

```diff
@@ -0,0 +1,199 @@
+
+typedef struct {
+    RzILOpBitVector *first;
+    RzILOpBitVector *second;
+} BVPair;
+
+RzILOpEffect *rz_msp430_dummy() {
+	return NOP(); 
+}
+
+RzILOpPure *get_destination(const Msp430Instruction *op) {
+    switch (op->dst_mode) {
+        case REG: {
+            RzILOpBitVector *reg = MSP430_GETR(op->dst);
+            if (op->word_sized & WORD_SIZED_READ) {
+                return reg;
+            }
+            return CAST(8, NULL, reg);
+        } 
+        default: return U16(0); // TODO : other addressing modes
+    }
+}
+
+BVPair get_destination_destructured(const Msp430Instruction *op) {
+    switch (op->dst_mode) {
+        case REG: {
+            RzILOpBitVector *reg = MSP430_GETR(op->dst);
+            BVPair res = {
+                .first = CAST(8, NULL, reg),
+                .second = CAST(8, NULL, SHIFTR(IL_FALSE, reg, UN8(8)))
+            };
+            return res;
+        }
+        default: {
+            BVPair dummy = {.first = U8(0), .second = U8(0)}; 
+            return dummy; // TODO : other addressing modes
+        }
+    }
+}
+
+RzILOpEffect *set_destination(const Msp430Instruction *op, RzILOpBitVector *old_value, RzILOpBitVector *new_value) {
+    switch (op->dst_mode) {
+        case REG: {
+            if (op->word_sized & WORD_SIZED_WRITE) {
+                return MSP430_SETR(op->dst, new_value);
+            }
+            // the general idea is: First we zero the lower byte through ANDing with 0xFF00
+            //                      Then we assign the lower byte to the (byte-length) result through ORing with 0x00<result>
+            // the overall effect is that only the lower byte is assigned, which is what byte-sized operations do
+            return MSP430_SETR(op->dst, OR(AND(old_value, U16(0xFF00)), UNSIGNED(16, new_value)));
+        } 
+        default: return U16(0); // TODO : other addressing modes
+    }
+}
+
+RzILOpBitVector *update_sr_nz_flags(const RzILOpBitVector *new_value, const RzILOpBitVector *old_sr_value) {
+    // the general idea is that we zero out the N and Z bits in the old SR value 
+    // (by ANDing with a mask of all 1s except in those positions)
+    // and then we OR that resulting value with the new bits in the relevant positions and all 0s everywhere else
+
+    // index of the N flag in the SR register: 2
+    RzILOpBool *n_flag_value = MSB(new_value);
+    RzILOpBitVector *n_or_mask = SHIFTL(IL_FALSE, UNSIGNED(16, n_flag_value), UN8(2));
+    
+    // index of the Z flag in the SR register: 1
+    RzILOpBool *z_flag_value = IS_ZERO(new_value);
+    RzILOpBitVector *z_or_mask = SHIFTL(IL_FALSE, UNSIGNED(16, z_flag_value), UN8(1));
+    
+    RzILOpBitVector *and_mask = UN16(~(1 << 2 | 1 << 1));
+    RzILOpBitVector *or_mask = OR(n_or_mask, z_or_mask);
+    
+    return OR(AND(old_sr_value, and_mask), or_mask);
+}
+
+RzILOpBitVector *update_sr_v_flag_rcc(const RzILOpBitVector *old_value, const RzILOpBool *old_carry, const RzILOpBitVector *old_sr_value) {
```

Comment:
```suggestion
RZ_OWN RzILOpBitVector *update_sr_v_flag_rcc(const RzILOpBitVector *old_value, const RzILOpBool *old_carry, const RzILOpBitVector *old_sr_value) {
```

It is only a draft. But don't forget to add those `RZ_OWN`, `RZ_BORROW` annotations later. They help quite a lot to force onceself to think about which component is responsible for what.